src/libraries/System.Text.Json/gen/JsonSourceGenerator.Roslyn3.11.cs

```diff
@@ -27,7 +28,7 @@ public sealed partial class JsonSourceGenerator : ISourceGenerator
         /// <param name="context"></param>
         public void Initialize(GeneratorInitializationContext context)
         {
-            context.RegisterForSyntaxNotifications(() => new SyntaxContextReceiver());
+            context.RegisterForSyntaxNotifications(() => new SyntaxContextReceiver(context.CancellationToken));
```

Comment:
@chsienki  is ok/correct to use this cancellation token here? is the delegate guaranteed to have a shorter lifetime than the CT for GeneratorInitializationContext ?

Reply:
No, you can't do this. On the second run through, the initialization isn't called again but the token will be something else, so you'll still be referring to the previously captured token. The lack of a token on the `GeneratorSyntaxContext` was an oversight.

In both compilers before 4.0 and currently we check the cancellation token before we visit each tree (https://sourceroslyn.io/#Microsoft.CodeAnalysis/SourceGeneration/SyntaxStore.cs,812c07816abb8b9e) 

It's obviously not as efficient as doing it per node visit but it will cancel out relatively quickly, at least.

Reply:
Good to know. I'll undo this